solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -381,7 +382,11 @@ public AssignmentException(String message, Throwable cause, boolean enableSuppre
 
   public interface AssignStrategy {
     List<ReplicaPosition> assign(SolrCloudManager solrCloudManager, AssignRequest assignRequest)
-        throws Assign.AssignmentException, IOException, InterruptedException;
+        throws AssignmentException, IOException, InterruptedException;
+    void verifyDeleteCollection(SolrCloudManager solrCloudManager, DocCollection collection)
```

Comment:
Maybe add default implementation here to not force empty implementation in legacy or other places?
I'd add javadoc at least for added methods...